lib/ecmascript.ts

```diff
@@ -221,594 +213,299 @@ function getIntlDateTimeFormatEnUsForTimeZone(timeZoneIdentifier) {
-        ES.ParseISODateTime(ES.ToString(relativeTo), { zoneRequired: false }));
-      if (ianaName) timeZone = ianaName;
-      if (z) {
-        offsetBehaviour = 'exact';
-      } else if (!offset) {
-        offsetBehaviour = 'wall';
-      }
-      if (!calendar) calendar = ES.GetISO8601Calendar();
-      calendar = ES.ToTemporalCalendar(calendar);
-    }
-    if (timeZone) {
-      timeZone = ES.ToTemporalTimeZone(timeZone);
-      let offsetNs = 0;
-      if (offsetBehaviour === 'option') offsetNs = ES.ParseOffsetString(ES.ToString(offset));
-      const epochNanoseconds = ES.InterpretISODateTimeOffset(
+export function ParseTemporalDateTimeString(isoString) {
+  return ParseISODateTime(isoString, { zoneRequired: false });
+}
+
+export function ParseTemporalDateString(isoString) {
+  return ParseISODateTime(isoString, { zoneRequired: false });
+}
+
+export function ParseTemporalTimeString(isoString) {
+  const match = PARSE.time.exec(isoString);
+  let hour, minute, second, millisecond, microsecond, nanosecond, calendar;
+  if (match) {
+    hour = ToInteger(match[1]);
+    minute = ToInteger(match[2] || match[5]);
+    second = ToInteger(match[3] || match[6]);
+    if (second === 60) second = 59;
+    const fraction = (match[4] || match[7]) + '000000000';
+    millisecond = ToInteger(fraction.slice(0, 3));
+    microsecond = ToInteger(fraction.slice(3, 6));
+    nanosecond = ToInteger(fraction.slice(6, 9));
+    calendar = match[15];
+  } else {
+    ({ hour, minute, second, millisecond, microsecond, nanosecond, calendar } = ParseISODateTime(isoString, {
+      zoneRequired: false
+    }));
+  }
+  return { hour, minute, second, millisecond, microsecond, nanosecond, calendar };
+}
+
+export function ParseTemporalYearMonthString(isoString) {
+  const match = PARSE.yearmonth.exec(isoString);
+  let year, month, calendar, referenceISODay;
+  if (match) {
+    let yearString = match[1];
+    if (yearString[0] === '\u2212') yearString = `-${yearString.slice(1)}`;
+    year = ToInteger(yearString);
+    month = ToInteger(match[2]);
+    calendar = match[3];
+  } else {
+    ({ year, month, calendar, day: referenceISODay } = ParseISODateTime(isoString, { zoneRequired: false }));
+  }
+  return { year, month, calendar, referenceISODay };
+}
+
+export function ParseTemporalMonthDayString(isoString) {
+  const match = PARSE.monthday.exec(isoString);
+  let month, day, calendar, referenceISOYear;
+  if (match) {
+    month = ToInteger(match[1]);
+    day = ToInteger(match[2]);
+  } else {
+    ({ month, day, calendar, year: referenceISOYear } = ParseISODateTime(isoString, { zoneRequired: false }));
+  }
+  return { month, day, calendar, referenceISOYear };
+}
+
+export function ParseTemporalTimeZoneString(stringIdent): Partial<{
+  year: number;
+  month: number;
+  day: number;
+  hour: number;
+  minute: number;
+  second: number;
+  millisecond: number;
+  microsecond: number;
+  nanosecond: number;
+  ianaName: string;
+  offset: any;
```

Comment:
Can we use a real type here?  There are only a few places where `any` is used, so seems like a good oppty to replace those with real types.

Reply:
Should this be treated as a string?